src/deploy/FlapperInit.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.8.0;
+
+import { DssInstance } from "dss-test/MCD.sol";
+import { FlapperInstance } from "./FlapperInstance.sol";
+
+interface FlapperUniV2Like {
+    function vat() external view returns (address);
+    function daiJoin() external view returns (address);
+    function spotter() external view returns (address);
+    function rely(address) external;
+    function file(bytes32, uint256) external;
+    function file(bytes32, address) external;
+}
+
+interface FlapperMomLike {
+    function setAuthority(address) external;
+}
+
+interface OracleWrapperLike {
+    function pip() external view returns (address);
+}
+
+interface PipLike {
+    function kiss(address) external;
+}
+
+struct FlapperUniV2Config {
+    uint256 hop;
+    uint256 want;
+    address pip;
+    uint256 bump;
+}
+
+library FlapperInit {
+    uint256 constant WAD = 10 ** 18;
+
+    function initFlapperUniV2(
+        DssInstance        memory dss,
+        FlapperInstance    memory flapperInstance,
+        FlapperUniV2Config memory cfg
+    ) internal {
+        FlapperUniV2Like flapper = FlapperUniV2Like(flapperInstance.flapper);
+        FlapperMomLike   mom     = FlapperMomLike(flapperInstance.mom);
+
+        // Sanity checks
+        require(flapper.vat()     == address(dss.vat),     "Flapper vat mismatch");
+        require(flapper.daiJoin() == address(dss.daiJoin), "Flapper daiJoin mismatch");
+        require(flapper.spotter() == address(dss.spotter), "Flapper spotter mismatch");
+
+        require(cfg.hop >= 5 minutes, "hop too low");
+        require(cfg.want >= WAD * 90 / 100, "want too low");
```

Comment:
Maybe:
```suggestion
        require(cfg.want >= WAD * 90 / 100 && cfg.want <= WAD, "want out of range");
```

Reply:
`want` can in theory be larger than `WAD`.
The idea was adding this for safety protection, in which case a too high `want` is not a safety concern.

Reply:
Hmmm fair enough, I thought more the idea was to force to a value which will potentially work. But in some weird situation a value of > WAD could also do.